utils.go

```diff
@@ -41,18 +41,31 @@ func generateJson(dirPath, filename string, data interface{}) (string, error) {
 
 	// Open the output file for writing.
 	filePath := path.Join(dirPath, filename)
-	out, err := os.OpenFile(filePath, os.O_CREATE | os.O_WRONLY, 0644)
+	out, err := os.OpenFile(filePath, os.O_CREATE|os.O_WRONLY, 0644)
 	if err != nil {
-		return "", errors.Wrapf(err,"unable to open %s for output", filePath)
+		return "", errors.Wrapf(err, "unable to open %s for output", filePath)
 	}
 	defer out.Close()
 
 	// Generate the output JSON.
 	encoder := json.NewEncoder(out)
 	err = encoder.Encode(data)
 	if err != nil {
-		return "", errors.Wrapf(err,"unable to marshal JSON to %s", filePath)
+		return "", errors.Wrapf(err, "unable to marshal JSON to %s", filePath)
 	}
 
 	return filePath, nil
 }
+
+func generateJsonContents(data interface{}) (*bytes.Buffer, error) {
+	var buffer bytes.Buffer
+	encoder := json.NewEncoder(&buffer)
+	encoder.SetIndent("", "    ")
+	err := encoder.Encode(data)
+	if err != nil {
+		return &buffer, errors.Wrap(err, "unable to marshal JSON")
+
+	}
+
+	return &buffer, nil
+}
```

Comment:
This function provides a way to marshal JSON to a byte buffer in order to make testing easier.